sdk/go/api_esc_extensions.go

```diff
@@ -0,0 +1,287 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return c.CheckEnvironmentYaml(ctx, org, yaml)
+}
+
+func (c *EscClient) CheckEnvironmentYaml(ctx context.Context, org, yaml string) (*CheckEnvironment, error) {
+	check, _, err := c.EscAPI.CheckEnvironmentYaml(ctx, org).Body(yaml).Execute()
+	var genericOpenApiError *GenericOpenAPIError
+	if err != nil && errors.As(err, &genericOpenApiError) {
+		model := genericOpenApiError.Model().(CheckEnvironment)
+		return &model, err
+	}
+
+	return check, err
+}
+
+func (c *EscClient) DecryptEnvironment(ctx context.Context, org, envName string) (*EnvironmentDefinition, string, error) {
+	env, resp, err := c.EscAPI.DecryptEnvironment(ctx, org, envName).Execute()
+
+	body, bodyErr := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, "", bodyErr
+	}
+
+	return env, string(body), err
+}
+
+func MarshalEnvironmentDefinition(env *EnvironmentDefinition) (string, error) {
+	var bs []byte
+	bs, err := yaml.Marshal(env)
+	if err == nil {
+		return string(bs), nil
+	}
+
+	return "", err
+}
+
+func mapValuesPrimitive(value any) any {
+	if val, ok := value.(*Value); ok {
+		return mapValuesPrimitive(val.Value)
+	}
+	if mapData, isMap := value.(map[string]Value); isMap {
+
+		output := make(map[string]any, len(mapData))
+		for k, v := range mapData {
+			output[k] = mapValuesPrimitive(v.Value)
+		}
+
+		return output
+	} else if sliceData, isSlice := value.([]any); isSlice {
+		for i, v := range sliceData {
+			sliceData[i] = mapValuesPrimitive(v)
+		}
+		return sliceData
+	}
+
+	return value
+}
+
+func mapValues(value any) any {
+	if val := getValue(getMapSafe(value)); val != nil {
+		val.Value = mapValues(val.Value)
+		return val
+	}
+	if mapData, isMap := value.(map[string]any); isMap {
+		output := map[string]Value{}
+		for key, v := range mapData {
+			value := mapValues(v)
+			if value == nil {
+				continue
+			}
+
+			if v, ok := value.(*Value); ok && v != nil {
+				output[key] = *v
+			} else {
+				output[key] = Value{
+					Value: value,
+				}
+			}
+		}
+		return output
+	} else if sliceData, isSlice := value.([]any); isSlice {
+		for i, v := range sliceData {
+			sliceData[i] = mapValues(v)
+		}
+		return sliceData
+	}
+
+	return value
+}
+
+func getValue(data map[string]any) *Value {
+	_, hasValue := data["value"]
+	_, hasTrace := data["trace"]
+	if hasValue && hasTrace {
+		return &Value{
+			Value:   mapValues(data["value"]),
+			Secret:  getBoolPtr(data, "secret"),
+			Unknown: getBoolPtr(data, "unknown"),
+			Trace:   getTrace(data["trace"].(map[string]any)),
+		}
+	}
+
+	return nil
+}
+
+func getTrace(data map[string]any) Trace {
+	def := getRange(getMapSafe(data["def"]))
+	base := getValue(getMapSafe(data["base"]))
+	if def != nil || base != nil {
+		return Trace{
+			Def:  def,
+			Base: base,
+		}
+	}
+
+	return Trace{}
+}
+
+func getMapSafe(data any) map[string]any {
+	if data != nil {
+		if val, ok := data.(map[string]any); ok {
+			return val
+		}
+
+		return nil
+	}
+
+	return nil
```

Comment:
This can be simplified:

```suggestion
	val, _ := data.(map[string]any)
	return val
```